test/EndToEnd/CLI.hs

```diff
@@ -1,118 +1,145 @@
+      removeFile fn
+
+mockDelegateState :: DelegateState
+mockDelegateState = Initialized (HeadId . fromBuiltin . unCurrencySymbol $ nonExistentHeadIdStub) (Open Nothing)
+
+auctionName :: AuctionName
+auctionName = "test"
+
+handleCliActionWithMockDelegates :: CliAction -> Runner ()
+handleCliActionWithMockDelegates action = do
+  delegateS <- newIORef mockDelegateState
+  handleCliAction (\_ -> pure ()) delegateS action
 
 bidderBuysTest :: Assertion
 bidderBuysTest = mkAssertion $ do
   let seller = Alice
       buyer1 = Bob
       buyer2 = Carol
 
-  handleCliAction $ Prepare seller
+  handleCliActionWithMockDelegates $ Prepare seller
 
   assertNFTNumEquals seller 1
 
-  handleCliAction $ AuctionAnounce "demo"
+  handleCliActionWithMockDelegates $ AuctionAnounce auctionName
+
+  terms <- auctionTermsFor auctionName
 
   waitUntil $ biddingStart terms
 
-  handleCliAction $ StartBidding "demo" [buyer1, buyer2]
+  handleCliActionWithMockDelegates $ StartBidding auctionName [buyer1, buyer2]
 
   assertNFTNumEquals seller 0
 
-  withActor buyer1 $ handleCliAction $ NewBid "demo" $ startingBid terms
-  withActor buyer2 $ handleCliAction $ NewBid "demo" $ startingBid terms + minimumBidIncrement terms
+  withActor buyer2 $ handleCliActionWithMockDelegates $ NewBid auctionName (startingBid terms) L1
 
   waitUntil $ biddingEnd terms
 
-  withActor buyer2 $ handleCliAction "demo"
+  withActor buyer2 $ handleCliActionWithMockDelegates $ BidderBuys auctionName
 
   assertNFTNumEquals seller 0
   assertNFTNumEquals buyer1 0
   assertNFTNumEquals buyer2 1
 
   waitUntil $ cleanup terms
-  handleCliAction $ Cleanup "demo"
+  handleCliActionWithMockDelegates $ Cleanup auctionName
+
+depositTest :: Assertion
+depositTest = mkAssertion $ do
+  let seller = Alice
+      buyer1 = Bob
+      buyer2 = Carol
+
+  handleCliActionWithMockDelegates $ Prepare seller
+
+  assertNFTNumEquals seller 1
+
+  handleCliActionWithMockDelegates $ AuctionAnounce auctionName
+
+  terms <- auctionTermsFor auctionName
+
+  withActor buyer1 $ handleCliActionWithMockDelegates $ MakeDeposit auctionName (fromJust $ intToNatural 10_000_000)
+  withActor buyer2 $ handleCliActionWithMockDelegates $ MakeDeposit auctionName (fromJust $ intToNatural 10_000_000)
+
+  assertUTxOsInScriptEquals Deposit terms 2
+
+  waitUntil $ biddingStart terms
+
+  handleCliActionWithMockDelegates $ StartBidding auctionName [buyer1, buyer2]
+
+  assertNFTNumEquals seller 0
+
+  withActor buyer2 $ handleCliActionWithMockDelegates $ NewBid auctionName (startingBid terms) L1
+
+  waitUntil $ biddingEnd terms
+
+  withActor buyer2 $ handleCliActionWithMockDelegates $ BidderBuys auctionName
+
+  assertNFTNumEquals seller 0
+  assertNFTNumEquals buyer1 0
+  assertNFTNumEquals buyer2 1
+
+  assertUTxOsInScriptEquals Deposit terms 1
+
+  withActor buyer1 $ handleCliActionWithMockDelegates $ BidderClaimsDeposit auctionName
+
+  assertUTxOsInScriptEquals Deposit terms 0
```

Comment:
Real ada checks are a bit tricky, because its hard to track fees, so we don't really have precise numbers to check.

Checking actor utxos would work (we will expect every losing bidder to have 1 utxo, as the claiming consolidates everything, and the winning bidder to have 2), but it is not a great test, as it would succeed even if the `BidderClaimsDeposit` tx sends the funds to the wrong account

Reply:
Fees can be taken from Tx or TxBody and taken into account.